### PR TITLE
Renamed BatchedDotOp to GpuBatchedDot

### LINKED ISSUE #3944: Rename `BatchedDotOp` to `GpuBatchedDot`
This would be more consistent with the usual naming scheme.
We should still be backward compatible, if possible with a warning.

### PR DESCRIPTION
**Pull request by Sentient07**
Fix #3944

### CODE REVIEW

File: theano/sandbox/cuda/blas.py [hunk 1]
```diff
@@ -216,7 +216,7 @@ def infer_shape(self, node, shapes):
         xshp, yshp = shapes
         return [xshp[:-1] + yshp[2:]]
 
-batched_dot = BatchedDotOp()
```
Comment (nouiz) [inline theano/sandbox/cuda/blas.py]: Also add "BatchedDotOp = GpuBatchedDot" to allow to unpickle old graph with the old class.

File: theano/sandbox/cuda/blas.py [hunk 2]
```diff
@@ -216,7 +216,8 @@ def infer_shape(self, node, shapes):
...
+batched_dot = GpuBatchedDot()
+BatchedDotOp() = GpuBatchedDot()
```
Comment (Sentient07) [inline theano/sandbox/cuda/blas.py]: @nouiz Added.

**Comment by nouiz** [comment]
the tests have this error:

```
  File "/home/travis/build/Theano/Theano/theano/sandbox/cuda/blas.py", line 220

    BatchedDotOp() = GpuBatchedDot()

                                   ^

SyntaxError: can't assign to function call
```

**Comment by Sentient07** [comment]
@nouiz. I'm sorry, forgot to remove the () while copy pasting. Corrected it now. :)

**Comment by nouiz** [comment]
I ran cuda/tests/test_blas.py and it passed. So merging. Thanks.

File: theano/sandbox/cuda/blas.py [hunk 3]
```diff
@@ -216,7 +216,8 @@ def infer_shape(self, node, shapes):
         xshp, yshp = shapes
         return [xshp[:-1] + yshp[2:]]
 
-batched_dot = BatchedDotOp()
+batched_dot = GpuBatchedDot()
+BatchedDotOp = GpuBatchedDot()
 """
 Call cublasSgemmBatched. Take 2 3d tensor as input.
 """
```
Comment (cooijmanstim) [inline theano/sandbox/cuda/blas.py]: I think two things went wrong here -- `batched_dot` is now separated from its docstring and `GpuBatchedDot` shouldn't be instantiated before assigning it to `BatchedDotOp`. @nouiz

Comment (nouiz) [inline theano/sandbox/cuda/blas.py]: I added a fix in https://github.com/Theano/Theano/pull/4045

thanks

**Pull request merged**